### PR TITLE
Add auto-refresh to page blocks

### LINKED ISSUE #466: Add support for auto-refresh of certain block-types
# Problem
Compose page blocks do not support auto-content refreshing and this forces users to refresh the browser or navigate away and back to the current page.

# Extending page block configurators
All page block configurators need to be extended to support auto-refresh settings, explicitly enable or disable manual refresh of a block.

Refresh interval can be expressed in seconds and must have lower limit at 5 seconds.

# Page refresh implementation
Since automatic refresh can interfere with user's workflow and can cause data loss, extra protection need to be implemented to prevent that.  

Image

- update 2022.3 Plan #4 @darh

### PR DESCRIPTION
**Pull request by EmmyMay**
Please take a look.

### CODE REVIEW

File: client/web/compose/src/components/PageBlocks/ChartBase.vue [hunk 1]
```diff
@@ -72,6 +80,12 @@ export default {
       const { namespaceID } = this.namespace
       return this.$ComposeAPI.recordReport({ namespaceID, ...nr })
     },
+
+    refresh () {
+      this.fetchChart({ force: true })
+    },
+
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/ChartBase.vue]: Remove empty line

File: client/web/compose/src/components/PageBlocks/CommentBase.vue [hunk 2]
```diff
@@ -227,7 +232,8 @@ export default {
       this.$store.dispatch('user/fetchUsers', userListID)
     },
 
-    reloadRecords () {
+    // reloadRecords
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/CommentBase.vue]: No need for this comment (doesn't really provide any info)

File: client/web/compose/src/components/PageBlocks/Configurator.vue [hunk 3]
```diff
@@ -80,14 +81,38 @@
...
+          v-if="block.options.refreshRate !== undefined"
+          :label="$t('general.refresh.label')"
+          :description="$t('general.refresh.description')"
         >
-          {{ $t('general.wrap') }}
-        </b-form-checkbox>
+          <b-col
+            cols="12"
+            sm="3"
+            class="pl-0"
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/Configurator.vue:103]: i think p-0 is the play here (not 100%)

File: client/web/compose/src/components/PageBlocks/RecordListBase.vue [hunk 4]
```diff
@@ -4,30 +4,18 @@
     v-bind="$props"
     :scrollable-body="false"
     v-on="$listeners"
+    @refreshBlock="refresh"
   >
     <template
-      v-if="showHeader"
-      #header
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordListBase.vue:10]: Please triple check if no other blocks used this slot, we need to handle that if they did

File: client/web/compose/src/components/PageBlocks/RecordOrganizerBase.vue [hunk 5]
```diff
@@ -421,6 +410,27 @@ export default {
       return this.$ComposeAPI.recordList({ namespaceID, moduleID, query, sort })
         .then(({ set }) => set.map(r => Object.freeze(new compose.Record(module, r))))
     },
+
+    // pull records
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordOrganizerBase.vue]: Remove this comment as well

**Review by Fajfa — CHANGES_REQUESTED** [review]

File: client/web/compose/src/components/PageBlocks/Configurator.vue (hunk 3 above)
Comment (EmmyMay) [reply client/web/compose/src/components/PageBlocks/Configurator.vue:103]: Not really. pl is okay. It gets too wide with p-0

**Pull request merged**